### DISCUSSION TITLE
(with-apollo) All queries get refetched on page refresh

### DISCUSSION THREAD

**Discussion by Bastaff** [vercel/next.js#17644] [Help]
As you can see the Post Feed and Sigle Post queries are both being loaded on page refresh no matter on which of the two pages you're on. This is the production build.

This is on loading index.js where the Post Feed is
![Screenshot_4](https://user-images.githubusercontent.com/55925093/95312134-38be1f80-0897-11eb-9582-f4049e8453f8.png)

This is on [slug].js where the Single Post is
![Screenshot_6](https://user-images.githubusercontent.com/55925093/95312153-3fe52d80-0897-11eb-9da7-3ae8928bf4fe.png)

Both queries seem to execute on page refresh on both these pages which is really bugging me and I can't understand why it's doing that. Caching and fetching data works flawlessly though.

index.js
```javascript
import App from '../components/App'
import InfoBox from '../components/InfoBox'
import Header from '../components/Header'
import SubmitPost from '../components/SubmitPost'
import PostList, {
  ALL_POSTS_QUERY,
  allPostsQueryVars,
} from '../components/PostList'
import { initializeApollo } from '../lib/apolloClient'
import Head from 'next/head'

const IndexPage = (props) => (
  <App>
    <Header />
    <InfoBox>Info.</InfoBox>
    <PostList/>
  </App>
)

export async function getStaticProps() {
 const apolloClient = initializeApollo()

  await apolloClient.query({
    query: ALL_POSTS_QUERY,
    variables: allPostsQueryVars,
  })

  return {
    props: {
      initialApolloState: apolloClient.cache.extract(),
    },
    revalidate: 1,
  }
}

export default IndexPage
```
[slug].js
```javascript
import App from '../components/App'
import Header from '../components/Header'
import Post, { POST_QUERY } from '../components/Post'
import { initializeApollo } from '../lib/apolloClient'
import { POST_SLUGS_QUERY } from '../components/queries/Queries'

const SinglePost = (props) => (
  <App>
    <Header />
    <Post />
  </App>
)


export async function getStaticPaths() {
  const apolloClient = initializeApollo();
  const { data } = await apolloClient.query({
    query: POST_SLUGS_QUERY
  })

  return {
    paths: data.postSlugs.map(post =>
      `/${post.slug}`) || [],
    fallback: true
  }
}


export async function getStaticProps({ params }) {
  const apolloClient = initializeApollo();

  await apolloClient.query({
    query: POST_QUERY,
    variables: {
      slug: params.slug,
    },
  });

  return {
    props: {
      initialApolloState: apolloClient.cache.extract(),
    },
    revalidate: 1,
  };
};


export default SinglePost
```
Is this normal?

**Comment by lfades** [comment]
Sounds like expected behavior to me 👍

**Reply by Dentoty** [reply]
Could you explain why this is expected?

**Reply by lotharvm** [reply]
I'd say this is not expected behaviour, as the first (server-side) query should be in Apollo's cache and therefore the client doesn't need to do the call again. You're sure the variables you're passing to the query are the same on both server and client?

**Reply by Bastaff** [reply]
@LotharVM Check the initialApolloState to the right in the screenshots and you can see that there are null prototype objects in the cache.
This could be one of the reasons why it's refetching on page refresh although when navigating through the website it's working fine.

Edit: 
Yes, index.js and the Postlist component are using the same Query and variables as are [slug].js and the Single Post component

**Reply by lfades** [reply]
@Bastaff Thank you for adding more details about the issue 🙏 

`apolloClient.cache.extract()` will be serialized to JSON and I'm expecting that to work without any issues 🤔 . After that `pages/_app.js` creates a store and merges the data that was returned in `initialApolloState`, the issues has to be somewhere in that flow.
